From 5559c66e9cd5d0b34390340374e66eab9933ec97 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 11 Jan 2021 15:42:33 +0800
Subject: [PATCH] usb: dwc3: gadget: disable lpm for rockchip platform

The rockchip DWC3 controllers are LPM capable, but we find
that UVC function compatibility issue on some Windows PCs
if enable LPM capable. And refer to the Logitech USB Cameras,
they're usually disable LPM capable. Considering that the
benefit form LPM capable is insignificant on Rochip platforms,
so we disable the LPM capable by default.

Change-Id: Ibdfb1131b0ecf5f6381f9eb1fc6bea7e7ecb8022
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc3/gadget.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index a028699b65bb..60152ffb09ae 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -3673,7 +3673,11 @@ int dwc3_gadget_init(struct dwc3 *dwc)
 	dwc->gadget.speed		= USB_SPEED_UNKNOWN;
 	dwc->gadget.sg_supported	= true;
 	dwc->gadget.name		= "dwc3-gadget";
+#ifdef CONFIG_ARCH_ROCKCHIP
+	dwc->gadget.lpm_capable		= false;
+#else
 	dwc->gadget.lpm_capable		= true;
+#endif
 
 	/*
 	 * FIXME We might be setting max_speed to <SUPER, however versions
-- 
2.35.3

